README.md

```diff
@@ -38,31 +38,21 @@ Or install it yourself as:
 
 ## Usage
 
-Get started by defining your own Analyzer, inheriting from `GraphQL::Metrics::Analyzer`.
-
-The following analyzer demonstrates a simple way to capture commonly used metrics sourced from key parts of your schema
-definition, the query document being served, as well as runtime query and resolver timings. In this toy example, all of
-this data is simply stored on the GraphQL::Query context, under a namespace to avoid collisions with other analyzers
-etc.
+Get started by writing a metrics processor class, inheriting from `GraphQL::Metrics::Processor`.
 
 What you do with these captured metrics is up to you!
 
 **NOTE**: If any non-`graphql-ruby` gem-related exceptions occur in your application during query document
-parsing and validation, **runtime metrics** for queries (like `query_duration`, etc.) as well as field
-resolver timings (like `resolver_timings`, `lazy_resolver_timings`) **may not be present** in the extracted `metrics` hash,
-even if you opt to collect them by using `GraphQL::Metrics::Analyzer` and `GraphQL::Metrics::Tracer`.
+parsing and validation, **runtime metrics** for queries (like `query_duration`) as well as field
+resolver timings (like `resolver_timings`, `lazy_resolver_timings`) **may not be present** in the extracted `metrics` hash.
 
 ### Define your own analyzer subclass
 
 ```ruby
-  class SimpleAnalyzer < GraphQL::Metrics::Analyzer
-    ANALYZER_NAMESPACE = :simple_analyzer_namespace
-
+  class SimpleProcessor < GraphQL::Metrics::Processor
     def initialize(query_or_multiplex)
       super
 
-      # `query` is defined on instances of objects inheriting from GraphQL::Metrics::Analyzer
-      ns = query.context.namespace(ANALYZER_NAMESPACE)
       ns[:simple_extractor_results] = {}
```

Comment:
If `ns` is now an attribute for the public interface, perhaps it can use a more meaningful name? `context` might be misleading since this is a separate object from the global context, but `namespace` isn't exactly descriptive either.

Reply:
oh... that's just wrong. I'll fix it